osgp/protocol-adapter-dlms/osgp-protocol-simulator-dlms/simulator/dlms-device-simulator/src/main/java/org/opensmartgridplatform/simulator/protocol/dlms/cosem/PowerQualityEventLog.java

```diff
@@ -0,0 +1,149 @@
+import org.opensmartgridplatform.simulator.protocol.dlms.cosem.processing.CaptureObjectDefinitionCollection;
+import org.opensmartgridplatform.simulator.protocol.dlms.cosem.processing.CosemDateTimeProcessor;
+import org.opensmartgridplatform.simulator.protocol.dlms.cosem.processing.UInteger8DataProcessor;
+
+@CosemClass(id = 7)
+public class PowerQualityEventLog extends ProfileGeneric {
+
+  private static final int CAPTURE_PERIOD = 0;
+  private static final int PROFILE_ENTRIES = 100;
+
+  private static final CaptureObjectDefinitionCollection CAPTURE_OBJECT_DEFINITIONS =
+      initCaptureObjects();
+
+  /**
+   * Only for cosem attribute definition, data remains untouched. Attribute data is gathered from
+   * {@link #bufferData}.
+   */
+  @CosemAttribute(
+      id = 2,
+      type = Type.ARRAY,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x08,
+      selector = {1, 2})
+  private final DataObject buffer;
+
+  /**
+   * Only for cosem attribute definition, data remains untouched. Attribute data is gathered from
+   * captureObjectDefinitions
+   */
+  @CosemAttribute(
+      id = 3,
+      type = Type.ARRAY,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x10)
+  private final DataObject captureObjects;
+
+  @CosemAttribute(
+      id = 4,
+      type = Type.DOUBLE_LONG_UNSIGNED,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x18)
+  public DataObject capturePeriod;
+
+  @CosemAttribute(
+      id = 5,
+      type = Type.ENUMERATE,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x20)
+  public DataObject sortMethod;
+
+  @CosemAttribute(
+      id = 6,
+      type = Type.STRUCTURE,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x28)
+  public DataObject sortObject;
+
+  /**
+   * Only for cosem attribute definition, data remains untouched. Attribute data is gathered from
+   * size of {@link #bufferData}
+   */
+  @CosemAttribute(
+      id = 7,
+      type = Type.DOUBLE_LONG_UNSIGNED,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x30)
+  private final DataObject entriesInUse;
+
+  @CosemAttribute(
+      id = 8,
+      type = Type.DOUBLE_LONG_UNSIGNED,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x38)
+  public DataObject profileEntries;
+
+  private final Calendar time;
+
+  public PowerQualityEventLog(final Calendar time) {
+    super("0.0.99.98.5.255");
+    this.time = time;
+
+    this.buffer = DataObject.newNullData();
+    this.captureObjects = DataObject.newNullData();
+    this.capturePeriod = DataObject.newInteger32Data(CAPTURE_PERIOD);
+    this.sortMethod = DataObject.newEnumerateData(SortMethod.FIFO.value());
+    this.sortObject = DataObject.newNullData();
+    this.entriesInUse = DataObject.newNullData();
+    this.profileEntries = DataObject.newInteger32Data(PROFILE_ENTRIES);
```

Comment:
The type for the `profileEntries` is `DOUBLE_LONG_UNSIGNED` while `newInteger32Data` returns a `DataObject` of type `DOUBLE_LONG`.
It looks like the method `newUInteger32Data` should be used here.